From e78f794c22e7825374d1d9bda37f649ed2f725d3 Mon Sep 17 00:00:00 2001
From: Natanael Copa <ncopa@alpinelinux.org>
Date: Mon, 14 Mar 2016 14:17:29 +0000
Subject: [PATCH] only enable the GNU libc strerror_r on GNU libc

We don't want use the non-standard GNU libc variant of strerror_r on
musl libc, or any libc other than GNU.
---
 src/os/linux/sigar_os.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/os/linux/sigar_os.h b/src/os/linux/sigar_os.h
index 29a2ba32..7e1ead7d 100644
--- a/src/os/linux/sigar_os.h
+++ b/src/os/linux/sigar_os.h
@@ -70,7 +70,7 @@ struct sigar_t {
 };
 
 #define HAVE_STRERROR_R
-#ifndef __USE_XOPEN2K
+#if !defined(__USE_XOPEN2K) && defined(__GLIBC__)
 /* use gnu version of strerror_r */
 #define HAVE_STRERROR_R_GLIBC
 #endif
